runner/runner.go

```diff
@@ -85,8 +93,17 @@ type Runner interface {
 	// check if cmd is well-formed, and reject it if not (leading to state FAILED)
 	// wait a very short period of time for cmd to finish
 	// Run may not wait indefinitely for cmd to finish. This is an async API.
-	Run(cmd *Command) (ProcessStatus, error)
+	Run(cmd *Command) ProcessStatus
 
 	// Status checks the status of run.
-	Status(run RunId) (ProcessStatus, error)
+	Status(run RunId) ProcessStatus
+
+	// Current status of all runs, running and finished.
```

Comment:
FWIW, I think it's fine if a Runner does throw out some statuses, so maybe say it's best effort for finished runs? WDYT?